x-pack/plugins/lens/public/xy_visualization_plugin/xy_suggestions.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { partition } from 'lodash';
+import { Position } from '@elastic/charts';
+import { XYArgs } from './xy_expression';
+import { SuggestionRequest, VisualizationSuggestion, TableColumn, TableSuggestion } from '../types';
+
+/**
+ * Generate suggestions for the xy chart.
+ *
+ * @param opts
+ */
+export function getSuggestions(
+  opts: SuggestionRequest<XYArgs>
+): Array<VisualizationSuggestion<XYArgs>> {
+  return Object.entries(opts.tables)
+    .filter(
+      ([_id, { isMultiRow, columns }]) =>
+        // We only render line charts for multi-row queries.
+        isMultiRow &&
+        // We require at least two columns: one for x and at least one for y
+        columns.length > 1 &&
+        // y columns must be numeric
+        columns.find(col => col.operation.dataType === 'number')
+    )
+    .map(([id, table]) => getSuggestionForColumns(id, table));
+}
+
+function getSuggestionForColumns(
+  id: string,
+  table: TableSuggestion
+): VisualizationSuggestion<XYArgs> {
+  const [buckets, values] = partition(
+    prioritizeColumns(table.columns),
+    x => x.operation.isBucketed
+  );
+
+  if (buckets.length >= 1) {
+    const [x, splitBy] = buckets;
+    return getSuggestion(id, x, values, splitBy);
+  } else {
+    const [x, ...yValues] = values;
+    return getSuggestion(id, x, yValues);
+  }
+}
+
+// This shuffles columns around so that the left-most column defualts to:
+// date, string, boolean, then number, in that priority. We then use this
+// order to pluck out the x column, and the split / stack column.
+function prioritizeColumns(columns: TableColumn[]) {
+  const columnSortOrder = {
+    date: 0,
+    string: 1,
+    boolean: 2,
+    number: 3,
+  };
+
+  return [...columns].sort(
+    (a, b) => columnSortOrder[a.operation.dataType] - columnSortOrder[b.operation.dataType]
+  );
+}
+
+function getSuggestion(
+  id: string,
+  xValue: TableColumn,
+  yValues: TableColumn[],
+  splitBy?: TableColumn
+): VisualizationSuggestion<XYArgs> {
+  const yTitle = yValues.map(v => v.operation.label).join(' & ');
```

Comment:
Variable naming; `v` is a column, so I would be consistent and use `col` as the temporary name